script.js

```diff
@@ -1,4 +1,76 @@
+// Rock Paper Scissor Game (Project 1)
+
+var currentGameMode = 'waiting for user name mode';
+var userName = '';
+
+var scoreUserWin = 0;
+var scoreCpuWin = 0;
+var scoreDraw = 0;
+
+// get random CPU pick from an array
+var cpuPick = function () {
+  var pickList = ['rock', 'scissor', 'paper'];
+  var randPick = pickList[Math.floor(Math.random() * pickList.length)];
+  console.log('computer picked ' + randPick + '!!!');
+  return randPick;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = '';
+  // do this if game mode is waiting for user...
+  if (currentGameMode == 'waiting for user name mode') {
+    // set the user name
+    userName = input;
+    // if user does not type anything
+    if (userName == '') {
+      return 'not a valid user name';
+    }
+    // now that we have user name, switch the mode
+    currentGameMode = 'game mode';
+    myOutputValue = 'Hello ' + userName + " let's play RSP";
+    return myOutputValue;
+  } if (currentGameMode == 'game mode') {
+    // game logic here
+    var userInput = input;
+    var randomCpuPick = cpuPick();
+    myOutputValue = 'Your pick is ' + input + '. CPU pick is ' + randomCpuPick + '!';
+    console.log(input);
+    console.log((input != 'rock' && input != 'paper' && input != 'scissor'));
+
+    // input not equal to rock, and not to paper, and not to scissor
+    if (input != 'rock' && input != 'paper' && input != 'scissor') {
+      return 'This is not a valid input, please enter only the word "rock", "paper", "scissor"';
+    }
+
+    // var compareGame = function (randomCpuPick, userInput) {}
+    if (randomCpuPick == userInput) {
+      scoreDraw = scoreDraw + 1;
+      console.log(scoreDraw);
+      return myOutputValue + " it's a tie. Scoreboard: " + 'D:' + scoreDraw + ' W:' + scoreUserWin + ' L:' + scoreCpuWin;
+    } if (randomCpuPick == 'rock' && userInput == 'paper') {
```

Comment:
Lots of repeated code here. Please refactor this to use `||` operator, e.g. if rock and paper OR paper and scissors OR scissors and stone THEN you win. Repeating code is **bad**. We will lose points and/or our job as SWEs if we don't understand how to avoid repeating code by using helper functions and/or conditional operators.